hathor/conf/get_settings.py

```diff
@@ -12,38 +12,29 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import importlib
 import os
-from types import ModuleType
 from typing import Optional
 
+from hathor import conf
 from hathor.conf.settings import HathorSettings as Settings
 
-_config_file: Optional[str] = None
+_settings_filepath: Optional[str] = None
 
 
 def HathorSettings() -> Settings:
-    """ Return configuration file namedtuple
-        Get the file from environment variable 'HATHOR_CONFIG_FILE'
-        If not set we return the config file of the mainnet
-    """
-    settings_module = get_settings_module()
-    settings = getattr(settings_module, 'SETTINGS')
-    assert isinstance(settings, Settings)
-    return settings
-
-
-def get_settings_module() -> ModuleType:
```

Comment:
A simple way to support the old settings scheme would be to keep this function, but make it return an `Optional[ModuleType]` so that `HathorSettings()` will try that before trying `get_settings_filepath()`. That way compatibility is not broken. That may also be where we log a warning that this mechanism is deprecated.

Reply:
Indeed, I followed your suggestion. I just had to change `get_settings_module()` so it wouldn't get the `hathor.conf.mainnet` module by default if `HATHOR_CONFIG_FILE` is not set. Instead, it returns `None` and it's `get_settings_filepath()`'s job to return `hathor/conf/mainnet.yml` as a default.

By doing this, we can still remove all python module files from our code, but keep compatibility for any third party using the old mechanism.

Done in https://github.com/HathorNetwork/hathor-core/pull/593/commits/e0c35ad26929f965144d4485c029e646cc6657b0

Reply:
That's great.

Reply:
I guess we can't remove because most custom settings inherit from mainnet. What do you think?

Reply:
Indeed, I talked with Jan about this in the part 2 PR: https://github.com/HathorNetwork/hathor-core/pull/597#issuecomment-1551961013

I guess this doesn't prevent this PR, part 1, from being merged, right? We just cannot merge part 2

Reply:
That's right. As we have the tests to confirm the yaml files are the same as the python modules, we are good.